services/src/main/java/org/keycloak/services/resources/admin/RealmAdminResource.java

```diff
@@ -763,6 +763,35 @@ public Response testLDAPConnection(@QueryParam("action") String action, @QueryPa
         return result ? Response.noContent().build() : ErrorResponse.error("LDAP test error", Response.Status.BAD_REQUEST);
     }
 
+    /**
+     * Test SMTP connection with current logged in user
+     *
+     * @param settings SMTP settings
+     * @return
+     * @throws Exception
+     */
+    @Path("testSMTPConnection")
+    @GET
+    @NoCache
+    public Response testSMTPConnection(@QueryParam("settings") String settings) throws Exception {
```

Comment:
I'd change this to a post so you can post a proper JSON document rather than encoding to a query param. Also, there's an issue here if authentication is enabled. Once you've saved the config, the password is not returned, so the config above would just have '********' the value for the password and not the real password.